src/main/java/edu/umich/its/cpm/MigrationInstanceService.java

```diff
@@ -53,19 +86,43 @@ public void runProcessingThread() throws InterruptedException {
 			// delay for 5 seconds
 			Thread.sleep(5000L);
 	        
-			// looping through
+			// looping through resource request
 			List<MigrationBoxFile> bFiles = fRepository.findNextNewMigrationBoxFile();
-			// process with the Box upload request
-			for(MigrationBoxFile bFile : bFiles)
+			if (bFiles != null && bFiles.size() > 0)
 			{
-				// mark the file as processed
-				fRepository.setMigrationBoxFileStartTime(bFile.getId(), new Timestamp(System.currentTimeMillis()));
+				// get right HttpContext object
+				HttpContext httpContext = Utils.login_become_admin(env);
```

Comment:
Will the login expire after a couple of hours?  Or does it keep getting renewed even when only accessed via batch?